dart/tool/java/generate_java.dart

```diff
@@ -407,9 +398,17 @@ class Method extends Member {
   void generateVmServiceForward(StatementWriter writer) {
     var consumerName = classNameFor(consumerTypeName);
     writer.addLine('if (consumer instanceof $consumerName) {');
-    var types = returnType.types.toList()
-      ..sort((t1, t2) => t1.name.compareTo(t2.name));
+    List<Type> types = returnType.types.map((ref) => ref.type).toList();
+    for (int index = 0; index < types.length; ++index) {
+      types.addAll(types[index].subtypes);
+    }
+    types.sort((t1, t2) => t1.name.compareTo(t2.name));
     for (var t in types) {
+
+      // TODO(danrubel) rename classes to prevent collision
+      // with common java types. e.g. "Class" --> "DartClass"
+      if (t.name == 'Class' || t.name == 'Error') continue;
+
       var jsonType = t.name;
       var responseName = classNameFor(t.elementTypeName);
       writer.addLine('  if (responseType.equals("$jsonType")) {');
```

Comment:
The doc writer class does have some methods which handle indenting and unindenting automatically, so the client doesn't have to explicitly send in indent strings. FYI for future changes :)